packages/framework/build.xml

```diff
@@ -72,13 +72,94 @@
 
     <target name="build" depends="build-deploy-part-1-db-independent,build-deploy-part-2-db-dependent" description="Builds application for production preserving your DB."/>
 
-    <target name="build-demo" depends="wipe,build-version-generate,composer-prod,redis-check,npm,dirs-create,assets,db-demo,grunt,error-pages-generate,warmup,product-search-recreate-structure,product-search-export-products,clean-redis-old" description="Builds application for production with clean demo DB."/>
+    <target name="build-demo" depends="production-protection" description="Builds application for production with clean demo DB.">
+        <if>
+            <equals arg1="${productionConfirmDestructiveAction}" arg2="y"/>
+            <then>
+                <phingcall target="wipe"/>
+                <phingcall target="build-version-generate"/>
+                <phingcall target="composer-prod"/>
+                <phingcall target="redis-check"/>
```

Comment:
you are missing `npm` here

Reply:
fixed

Reply:
originally, the `npm` was called after `redis-check`, I think you should not change that unless you have some reason for that :innocent:

Reply:
I'm not sure if phingcall will not call duplicate targets more then once.

Reply:
@henzigo I do not understand what you mean

Reply:
Oh, I will show you an example:

Look about those commands:
```
    <target name="db-check-mapping" depends="clean-redis">
    ...
    </target>

    <target name="db-check-schema" depends="clean-redis,db-check-mapping">
       ...
    </target>
```

If you will have command `db-check` that has to depend on those two commands then `clean-redis` will be called only once. 

But if you will call `php phing db-check-mapping db-check-schema` then `clean-redis` will be called twice and I'm not sure which of those rules will be applied if you will use phingcall

Reply:
@henzigo I understand now. I am afraid that security of production is more important then little longer build, do you agree with me?

Reply:
But you can easily check that in production-protection command (and throw error if you will not confirm that) and then add only depends on that target without change any of those commands, so you will use depends and you will have protection.

Reply:
Hi @TomasLudvik, I do agree with @henzigo here, it would be easier to move the logic to the protection target, and then let the other targets be protected by adding `production-protection` target as their dependency, what do you say? The thing is, it is not just about longer build - sometimes running one target multiple times might be harmful. And moreover, repeating this condition
```
<if>
   <equals arg1="${production.confirm.action}" arg2="y"/>
   <then>
...
```
in multiple targets is DRY violaion :slightly_frowning_face: